### ISSUE TITLE
Why is gsl::span<> not compatible to c++2a std::span<>?

### ISSUE THREAD

**Issue by ClausKlein** [martinmoene/gsl-lite#171]
```c++
|| /opt/local/bin/ccache /opt/local/bin/clang++    -std=c++2a -Wpedantic -g   -o CMakeFiles/use_span.dir/use_span.cpp.o -c /Users/clausklein/Workspace/cpp/samples/use_span.cpp
/Users/clausklein/Workspace/cpp/samples/use_span.cpp|22 col 5| error: too many template arguments for class template 'span'
||     span<int, COUNT> av = arr;
||     ^         ~~~~~~
/usr/local/include/gsl/gsl-lite.hpp|1563 col 38| note: template is declared here
||     template< class U > friend class span;
||     ~~~~~~~~~~~~~~~~~~~              ^
/Users/clausklein/Workspace/cpp/samples/use_span.cpp|33 col 5| error: too many template arguments for class template 'span'
||     span<int, COUNT> av = arr;
||     ^         ~~~~~~
/usr/local/include/gsl/gsl-lite.hpp|1563 col 38| note: template is declared here
||     template< class U > friend class span;
||     ~~~~~~~~~~~~~~~~~~~              ^
|| 2 errors generated.
```

**Comment by ClausKlein** [comment]
My used sample program:

```c++
// A span is a non-owning view over a contiguous sequence of objects, the
// storage of which is owned by some other object.

#define GSL_THROW_ON_CONTRACT_VIOLATION 1
#include <gsl/gsl> // C++14

#include <array> // C++11
#include <span>  // C++20
using gsl::at;
using std::span;

const size_t COUNT(7);

// Pointers should only refer to single objects!
void use(int *ptr, int value) { *ptr = value; }

// Use a gsl::span and range-for:
void f1a()
{
    int arr[COUNT];
    // FIXME gsl::span<int, COUNT> av = arr;
    span<int, COUNT> av = arr;
    int i = 0;
    for (auto &e : av)
        e = i++;
}

// Use gsl::at() for access:
void f2()
{
    int arr[COUNT];
    // FIXME gsl::span<int, COUNT> av = arr;
    span<int, COUNT> av = arr;
    for (int i = 0; i < COUNT; ++i) {
        at(arr, i) = i;
    }
}

// BETTER: use span in the function declaration
void f(span<int> a)
{
    if (a.size() < 2)
        return;

    int n = a[0]; // OK
    if (!n)
        return;

    span<int> q = a.subspan(1); // OK
    f(q);

    if (a.size() < 6)
        return;

    a[4] = 1; // OK

    a[a.size() - 1] = 2; // OK

    use(a.data(), 3); // OK
}

int main()
{
    std::array<int, COUNT> a{{1, 2, 3, 4, 5, 6}};
    f(a);
    f1a();
    f2();
}
```

**Comment by martinmoene** [comment]
See [milestone 7](https://github.com/martinmoene/gsl-lite/milestone/7).

**Comment by mbeutel** [comment]
Closed as duplicate of #153.
